lisp/ess-custom.el

```diff
@@ -646,211 +646,345 @@ regardless of where in the line point is when the TAB command is used."
-                                    arg2)
 
 If a list of the form '(N) where N is a number, the statements
 are indented at the previous line indentation + N characters:
 
-  a <- some.function(other.function(
-     arg1,
-     arg2)
+  object <- some_function(other_function(arg1,
+      arg2,
+      arg3)
 
 
-If a number N, the statement are alligned at the beginning of
-function call + N characters:
+If nil, the statements are indented at the opening delimiter
+following foo:
 
-  a <- some.function(other.function(
-                       arg1,
-                       arg2)
+  object <- some_function(other_function(arg1,
+                                         arg2,
+                                         arg3)
 
 
-For inner function arguments the behavior is unchanged:
+You can refer to ess-indent-level by setting this parameter to t
+or '(t) instead of N or '(N).
+")
+
+(defvar ess-offset-arguments-newline '(t)
+  "Extra indent for function arguments or bracket indexing when (
+or [ is followed by a new line.
+
+If a number N, the statement are aligned at the beginning of the
+closest function call + N characters:
 
-  some.function(arg1,
-                arg2=other.function(a,
-                  b,
+  object <- some_function(other_function(
+                              arg1,
+                              arg2)
 
-Set `ess-arg-function-offset' to nil if you want:
 
-  some.function(arg1,
-                arg2=other.function(a,
-                                    b,
+If a list of the form '(N) where N is a number, the statements
+are indented at the previous line indentation + N characters:
+
+  object <- some_function(other_function(
+      arg1,
+      arg2)
 
-and
 
-some.function(arg1,
-              arg2=other.function(
-                     a,
-                     b,
+If nil, the statements are indented at the opening delimiter
+following foo:
 
+  object <- some_function(other_function(
+                                         arg1,
+                                         arg2)
+
+
+You can refer to ess-indent-level by setting this parameter to t
+or '(t) instead of N or '(N).
 ")
 
-;;added rmh 2Nov97 at request of Terry Therneau
-(defvar ess-close-brace-offset 0
-  "Extra indentation for closing braces.")
-
-(defvar ess-close-paren-offset 0
-  "Extra indentation for closing parenthesis.
-When a number, adjustment is made with respect to the opening
-parenthesis taking into account the value of
-`ess-arg-function-offset-new-line'. When set to 0 the closing
-parenthesis is indented as follows:
-
-some.function(arg1,
-              arg2
-              )
-
-When this variable is a list of the form '(N), N being a number,
-adjust with respect to the indentation of the line containing an
-opening parenthesis. For example, when set to '(0), the closing
-parenthesis is indented as follows:
-
-{
-    some.function(arg1,
-                  arg2 = X
-    )
-}
+(defvar ess-offset-arguments-closing 0
```

Comment:
personally I think _all_ offset delimiters are superfluous. I included them because they are a legacy of the past. But I think they are unnecessary especially now that we have the flexibility offered by the three main offsets settings.

Should I remove them all?